locations/spiders/circle_k_pl.py

```diff
@@ -0,0 +1,114 @@
+            item["city"] = location["/sites/{siteId}/addresses"]["PHYSICAL"]["city"]
+            item["state"] = location["/sites/{siteId}/addresses"]["PHYSICAL"]["state"]
+            item["postcode"] = location["/sites/{siteId}/addresses"]["PHYSICAL"]["postalCode"]
+            item["country"] = location["/sites/{siteId}/addresses"]["PHYSICAL"]["country"]
+            item["operator"] = location["/sites/{siteId}/business-info"]["companyName"]
+
+            if phones := location["/sites/{siteId}/contact-details"]["phones"]:
+                item["phone"] = phones.get("WOR")
+                item["extras"]["fax"] = phones.get("FAX")
+
+            if emails := location["/sites/{siteId}/contact-details"]["emails"]:
+                item["email"] = "; ".join(emails["DN"])
+
+            services = [s["name"] if s["state"] == "ENABLED" else None for s in location["/sites/{siteId}/services"]]
+            apply_yes_no(Extras.WHEELCHAIR, item, "EU_ACCESSIBLE_FACILITIES" in services)
+            apply_yes_no(Extras.ATM, item, "EU_ATM" in services)
+            apply_yes_no(Extras.BABY_CHANGING_TABLE, item, "EU_BABY_CHANGING" in services)
+            apply_yes_no(Extras.CAR_WASH, item, "EU_CARWASH" in services or "EU_CARWASH_JETWASH" in services)
+            apply_yes_no("sells:lottery", item, "EU_LOTTO" in services)
+            apply_yes_no(Extras.SELF_CHECKOUT, item, "EU_SELF_CHECKOUT" in services)
+            apply_yes_no(Extras.SHOWERS, item, "EU_SHOWER" in services)
+            apply_yes_no(Extras.TOILETS, item, "EU_TOILETS_BOTH" in services)
+            apply_yes_no(Extras.WIFI, item, "EU_WIFI" in services)
+
+            shop = None
+            if location["/sites/{siteId}/business-info"]["chainConvenience"]:
+                shop = item.deepcopy()
+                shop["ref"] += "-shop"
+                apply_category(Categories.SHOP_CONVENIENCE, shop)
+                shop["opening_hours"] = parse_opening_hours(
+                    location["/sites/{siteId}/opening-info"], "openingTimesStore"
+                )
+
+            apply_category(Categories.FUEL_STATION, item)
+            item["opening_hours"] = parse_opening_hours(location["/sites/{siteId}/opening-info"], "openingTimesFuel")
+
+            for fuel in location["/sites/{siteId}/fuels"]:
+                if fuel["name"] not in FUELS_MAP:
+                    self.crawler.stats.inc_value(
+                        "atp/circle_k_pl/fuels_map/missing/{}/{}".format(fuel["name"], fuel["displayName"])
+                    )
+                if fuel := FUELS_MAP.get(fuel["name"]):
+                    apply_yes_no(fuel, item, True)
+
+            # Follow redirect to collect real url
+            yield Request(
+                "https://www.circlek.pl/station/{}".format(item["ref"]),
+                callback=self.parse_webpage,
+                cb_kwargs={"fuel": item, "shop": shop},
+            )
```

Comment:
> 'downloader/request_count': 820,

@Cj-Malone @iandees, question: the spider makes a Request for every POI to get a better urls from redirect, that's quite a lot since we get all the data from json blob on first request. Do we want to remove this part of the code to ease load on a website?